inc/default-styles.php

```diff
@@ -433,97 +310,119 @@ static function widget_style_fields( $fields ) {
 		return $fields;
 	}
 
-	static function row_style_attributes( $attributes, $args ) {
-		if ( ! empty( $args['row_stretch'] ) ) {
+	static function row_style_attributes( $attributes, $style ) {
+		if ( ! empty( $style['row_stretch'] ) ) {
 			$attributes['class'][]           = 'siteorigin-panels-stretch';
-			$attributes['data-stretch-type'] = $args['row_stretch'];
+			$attributes['data-stretch-type'] = $style['row_stretch'];
 			wp_enqueue_script( 'siteorigin-panels-front-styles' );
 		}
 
-		if ( ! empty( $args['class'] ) ) {
-			$attributes['class'] = array_merge( $attributes['class'], explode( ' ', $args['class'] ) );
+		if ( ! empty( $style['class'] ) ) {
+			$attributes['class'] = array_merge( $attributes['class'], explode( ' ', $style['class'] ) );
 		}
 
-		if ( ! empty( $args['row_css'] ) ) {
-			preg_match_all( '/^(.+?):(.+?);?$/m', $args['row_css'], $matches );
+		if ( ! empty( $style['background_display'] ) && ! empty( $style['background_image_attachment'] ) ) {
 
-			if ( ! empty( $matches[0] ) ) {
-				for ( $i = 0; $i < count( $matches[0] ); $i ++ ) {
-					$attributes['style'] .= $matches[1][ $i ] . ':' . $matches[2][ $i ] . ';';
-				}
+			if ( $style['background_display'] == 'parallax' || $style['background_display'] == 'parallax-original' ) {
+				wp_enqueue_script( 'siteorigin-panels-front-styles' );
 			}
-		}
 
-		if ( ! empty( $args['background'] ) ) {
-			$attributes['style'] .= 'background-color:' . $args['background'] . ';';
+			$url = wp_get_attachment_image_src( $style['background_image_attachment'], 'full' );
```

Comment:
Oh rad, and it turns out background images can have a srcset equivalent http://stackoverflow.com/questions/26801745/is-there-a-srcset-equivalent-for-css-background-image - I'll log this as a separate issue, though.